### PR TITLE
Update readme post extraction

### PR DESCRIPTION
**Pull request by ross**
/cc https://github.com/octodns/octodns/issues/622 for the road to 1.0

### CODE REVIEW

**Comment by ross** [comment]
[Rendered](https://github.com/octodns/octodns/blob/update-readme-post-extraction/README.md#providers)

File: README.md [hunk 1]
```diff
@@ -12,21 +12,28 @@ The architecture is pluggable and the tooling is flexible to make it applicable
...
+  * [Config](#config)
+  * [Noop](#noop)
+  * [Making changes](#making-changes)
+  * [Workflow](#workflow)
+  * [Bootstrapping config files](#bootstrapping-config-files)
+- [Providers](#providers)
+  * [Updating to use extracted providers](#updating-to-use-extracted-providers)
+- [Sources](#sources)
+    + [Notes](#notes)
+- [Compatibilty & Compliance](#compatibilty---compliance)
```
Comment (viranch) [inline README.md:26]: Click-and-jump is not working on this anchor link.

File: README.md [hunk 2]
```diff
@@ -12,21 +12,28 @@ The architecture is pluggable and the tooling is flexible to make it applicable
...
 - [Custom Sources and Providers](#custom-sources-and-providers)
 - [Other Uses](#other-uses)
-  - [Syncing between providers](#syncing-between-providers)
-  - [Dynamic sources](#dynamic-sources)
+  * [Syncing between providers](#syncing-between-providers)
+  * [Dynamic sources](#dynamic-sources)
 - [Contributing](#contributing)
 - [Getting help](#getting-help)
-- [Related Projects & Resources](#related-projects--resources)
+- [Related Projects & Resources](#related-projects---resources)
```
Comment (viranch) [inline README.md:36]: Click-and-jump is not working on this anchor link.

Comment (ross) [reply README.md:36]: Hrm, used one of those auto-generator deals, but I guess it didn't support the &. Will rejigger things to get them happy when I get a chance to sit down with it.

**Review by viranch — APPROVED** [review]
Looks good after above minor comments are resolved. Can also skip them and fix later if they're not critical.

**Pull request merged**